toolkit/tools/specreader/specreader.go

```diff
@@ -82,13 +84,55 @@ func main() {
 	toolchainRPMs, err := schedulerutils.ReadReservedFilesList(*toolchainManifest)
 	logger.PanicOnError(err, "Unable to read toolchain manifest file '%s': %s", *toolchainManifest, err)
 
-	err = parseSPECsWrapper(*buildDir, *specsDir, *rpmsDir, *srpmsDir, *existingToolchainRpmDir, *distTag, *output, *workerTar, toolchainRPMs, *workers, *runCheck)
+	// A parse list may be provided, if so only parse this subset.
+	// If none is provided, parse all specs.
+	specListSet, err := parseSpecListFile(*specListFile)
 	logger.PanicOnError(err)
+
+	// Convert specsDir to an absolute path
+	specsAbsDir, err := filepath.Abs(*specsDir)
+	logger.PanicOnError(err, "Unable to get absolute path for specs directory '%s': %s", *specsDir, err)
+
+	err = parseSPECsWrapper(*buildDir, specsAbsDir, *rpmsDir, *srpmsDir, *existingToolchainRpmDir, *distTag, *output, *workerTar, specListSet, toolchainRPMs, *workers, *runCheck)
+	logger.PanicOnError(err)
+}
+
+// parsePackListFile will parse a list of packages to pack if one is specified.
+// Duplicate list entries in the file will be removed.
+func parseSpecListFile(specListFile string) (specListSet map[string]bool, err error) {
```

Comment:
I think we're duplicating code from `srpmpacker.go`. Let's move it to a shared module, so there's only one copy.

Reply:
Fixed